src/views/TabView/TabView.js

```diff
@@ -110,54 +110,56 @@ class TabView extends PureComponent<void, Props, void> {
 
   _renderTabBar = (props: *) => {
     const {
-      tabBarOptions,
-      tabBarComponent: TabBarComponent,
       animationEnabled,
+      navigation,
+      router,
+      screenProps,
+      tabBarComponent: TabBarComponent,
+      tabBarOptions,
     } = this.props;
     if (typeof TabBarComponent === 'undefined') {
       return null;
     }
+
+    const { state } = navigation;
+    const options = router.getScreenOptions(
+      this.props.childNavigationProps[state.routes[state.index].key],
+      screenProps || {}
+    );
+
+    const tabBarVisible = options.tabBarVisible === null
```

Comment:
I just saw @matthamil's comment [above](https://github.com/react-community/react-navigation/pull/2101#discussion_r126574026) recommending this originally. To be clear, this code was simply moved from lower down and needs to check for `undefined` also, otherwise the tabs all disappear if `tabBarVisible` is not explicitly set.